Mathlib/Data/List/Triwise.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2025 Joseph Myers, Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers, Yaël Dillies
+-/
+import Aesop
+import Mathlib.Tactic.Lemma
+import Mathlib.Tactic.MkIffOfInductiveProp
+
+/-!
+# Triwise predicates on list.
```

Comment:
Is "triwise" a word? I'd expect to see it written `tripletwise`, at least in a docstring if that's too long for a name. After all, we say "pairwise" not "biwise".

Reply:
OK, I've renamed it to `Tripletwise` everywhere. (The original name was just taken from Yaël's suggestion on Zulip.)

As for `Set` and `_root_` definitions, they're not relevant for my current use case but they seem reasonable to add separately anyway. (There would be other geometrical uses for such definitions; for example, "a set of points is collinear iff they are tripletwise collinear".)

Reply:
`triple` is the usually mathematical term, not `triplet`, no?

Reply:
Similarly below, it's definitely not `tuplet`, just `tuple`.

Reply:
You're quite right, musical naming was sneaking in there.

Reply:
Renamed now to `Triplewise`.

We have `tuple`s in mathlib, they're `Fin n → α`. (I think polygons built on tuples will end up being a more convenient way of talking about cyclic polygons than something built on lists, and while such things could be defined in terms of either `Triplewise` (all points on same circle and all triples in the same cyclic order) or `Quadruplewise` (all quadruples are cyclic quadrilaterals), more convenient definitions in that case probably won't be based on those predicates.)